bridge/Mixer.cpp

```diff
@@ -768,32 +768,45 @@ std::unordered_set<std::string> Mixer::getEndpoints() const
     return endpoints;
 }
 
-bool Mixer::getEndpointInfo(const std::string& endpointId, api::ConferenceEndpoint& endpoint)
+std::unordered_set<size_t> Mixer::getActiveTalkers()
+{
+    return _engineMixer.getActiveTalkers();
+}
+
+bool Mixer::getEndpointInfo(const std::string& endpointId,
+    api::ConferenceEndpoint& endpoint,
+    const std::unordered_set<size_t>& activeTalkers)
 {
     std::lock_guard<std::mutex> locker(_configurationLock);
     const auto audio = _audioStreams.find(endpointId);
     endpoint.id = endpointId;
-    endpoint.hasAudio = false;
-    endpoint.isActiveSpeaker = false;
+    endpoint.isDominantSpeaker = false;
+    endpoint.isActiveTalker = false;
+    bool foundAudio = false;
     if (audio != _audioStreams.cend())
     {
-        endpoint.hasAudio = true;
         if (audio->second)
         {
-            endpoint.isActiveSpeaker =
-                (audio->second) && audio->second->endpointIdHash == _engineMixer.getDominantSpeakerId();
+            foundAudio = true;
+            endpoint.isDominantSpeaker = audio->second->endpointIdHash == _engineMixer.getDominantSpeakerId();
+            if (activeTalkers.find(audio->second->endpointIdHash) != activeTalkers.end())
+            {
+                endpoint.isActiveTalker = true;
+            }
+            else
+            {
+                endpoint.isActiveTalker = false; // endpoint.isDominantSpeaker;
+            }
 
             auto transport = audio->second->transport;
             endpoint.iceState = transport->getIceState();
             endpoint.dtlsState = transport->getDtlsState();
         }
     }
 
-    endpoint.hasVideo = _videoStreams.find(endpointId) != _videoStreams.cend() ? true : false;
-    endpoint.isBundled = _bundleTransports.find(endpointId) != _bundleTransports.cend() ? true : false;
-    endpoint.isRecording = _recordingStreams.find(endpointId) != _recordingStreams.cend() ? true : false;
+    bool foundVideo = _videoStreams.find(endpointId) != _videoStreams.cend() ? true : false;
```

Comment:
No need for ?
The expression is already true or false.

Reply:
Fixed and inlined.